### PR TITLE
refactor: simplify gas_counter

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: runtime/near-vm-logic/src/gas_counter.rs [hunk 1]
```diff
@@ -16,12 +16,7 @@ thread_local! {
 
 #[cfg(all(feature = "costs_counting", feature = "protocol_feature_evm"))]
 pub fn reset_evm_gas_counter() -> u64 {
-    let mut ret = 0;
-    EVM_GAS_COUNTER.with(|f| {
-        ret = *f.borrow();
-        *f.borrow_mut() = 0;
-    });
-    ret
+    EVM_GAS_COUNTER.with(|f| f.replace(0))
```
Comment (matklad) [inline runtime/near-vm-logic/src/gas_counter.rs:19]: [Jones's trick](http://giphygifs.s3.amazonaws.com/media/MS0fQBmGGMaRy/giphy.gif) can simplify a lot of things. Many types have `.replace` method: https://doc.rust-lang.org/stable/std/?search=replace. There's also `std::mem::replace`.

File: runtime/near-vm-logic/src/gas_counter.rs [hunk 2]
```diff
@@ -95,56 +90,45 @@ impl GasCounter {
...
-
-    #[cfg(feature = "costs_counting")]
     #[inline]
     fn update_profile_action(&mut self, action: ActionCosts, value: u64) {
-        match &self.profile {
-            Some(profile) => profile.add_action_cost(action, value),
-            None => {}
-        };
+        if cfg!(feature = "costs_counting") {
+            match &self.profile {
```
Comment (bowenwang1996) [inline runtime/near-vm-logic/src/gas_counter.rs:125]: is it better to use `if let` here

Comment (matklad) [reply runtime/near-vm-logic/src/gas_counter.rs:125]: Yeah, let me refactor it as well while I am at it.

My rule of thumb is "use if let iff there's no else". With else, `match` actually reads better and I like that ignoring (`_`) is explicit.

**Review by frol — APPROVED** [review]

**Comment by frol** [comment]
I have already restarted the `db-migration` CI test 4 times and it still fails. I don't expect this change to have any effect on the test, but it seems @matklad is super unlucky :smile:

**Comment by matklad** [comment]
ping @evgenykuzyakov, I *think* this is blocked on review from you (although I might be missing some process here). If this looks good, I'll rebase.

**Comment by frol** [comment]
@matklad please, rebase, I will merge it since Evgeny is out of the office.

**Comment by matklad** [comment]
Ah! rebased @frol !

**Comment by matklad** [comment]
And now rebased *properly*

**Pull request merged**